Doc/Manual/Fortran.html

```diff
@@ -0,0 +1,2275 @@
+</p>
+
+
+<p>
+When exception handling code is used, SWIG generates a few internal data structures as well as two externally accessible symbols with external C linkage (<code>ierr</code> and <code>get_serr</code>). Fortran bindings are generated to make the integer and function accessible from the Fortran module.
+</p>
+
+
+<p>
+The names of the integer and string accessor have C linkage and thus must be unique in a compiled program. Since other translation units might have symbols that share the default exception handling names, the user can provide custom names before including the exception handling file:
+</p>
+
+
+<div class="code"><pre><code>#define SWIG_FORTRAN_ERROR_INT my_ierr
+#define SWIG_FORTRAN_ERROR_STR get_my_serr
+%include &lt;std_except.i&gt;</code></pre></div>
+
+
+<p>
+If you&rsquo;re linking multiple modules together (using %import or otherwise), only one of those modules should define the error integer and accessor by including <code>&lt;std_except.i&gt;</code> or <code>&lt;exception.i&gt;</code>. Every other module needs to add
+</p>
+
+
+<div class="code"><pre><code>%include &lt;extern_exception.i&gt;</code></pre></div>
+
+
+<p>
+before any other module is <code>%import</code>ed (or any other exception-related source files are <code>%include</code>d). This inserts the correct exception macros in the wrapper code and <em>declares</em> (but does not define) the external-linkage error function and variable. You must also ensure the <code>SWIG_FORTRAN_ERROR_INT</code> macro is correctly defined before this include if it&rsquo;s being used upstream.
+</p>
+
+
+<p>
+If you forget to make the above inclusion and an <code>%import</code>ed module loads <code>exception.i</code>, a SWIG error will be displayed with a reminder of what to do. If <em>all</em> of your modules declare <code>extern_exception.i</code>, the program will fail to link due to the undefined symbols.
+</p>
+
+<!-- ###################################################################### -->
+<H2><a name="Fortran_provided_typemaps">Provided typemaps</a></H2>
+
+<p>
+There are many ways to make C++ data types interact more cleanly with Fortran types. For example, it&rsquo;s common for C++ interfaces take a <code>std::string</code> when they&rsquo;re typically called with string literals: the class can be implicitly constructed from a <code>const char *</code> but can also accept a <code>std::string</code> if needed. Since Fortran has no implicit constructors, passing a string argument would typically require declaring and instantiating a class for that parameter. To mitigate this annoyance, special typemaps are provided that transparently convert between Fortran types and C++ types while
+</p>
```

Comment:
This looks unfinished!

Reply:
Yep. If you've ever seen the movie *Up!* you might recall [how the dog gets distracted midsentence by a squirrel](https://giphy.com/gifs/funny-dogsis-rGLyjVneXGNCE) -- that happens embarrassingly frequently to me for documentation. I will fix.